plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/catalog/file/TestIcebergFileCreateTableFailureMetadataCleanedUp.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.iceberg.catalog.file;
+
+import org.testng.annotations.Test;
+
+import java.io.IOException;
+
+@Test(singleThreaded = true)
+public class TestIcebergFileCreateTableFailureMetadataCleanedUp
+        extends BaseIcebergFileCreateTableFailureTest
+{
+    private static final String EXCEPTION_MESSAGE = "Test-simulated metastore unsupported operation exception";
+
+    @Override
+    protected void throwCreateTableException()
+    {
+        throw new UnsupportedOperationException(EXCEPTION_MESSAGE);
```

Comment:
Why UnsupportedOperationException?
Would it make sense to simulate some other exception?